PWGHF/HFL/Tasks/taskSingleMuon.cxx

```diff
@@ -227,6 +230,39 @@ struct HfTaskSingleMuonSelection {
     } // end loop over muon tracks
   }
 
+  template <uint32_t TEventFillMap, uint32_t TMuonFillMap, typename TEvent, typename TMuons>
+  void runMuonSelLite(TEvent const& event, aod::BCs const& bcs, TMuons const& tracks)
```

Comment:
Why do you need a new function which seems to differ from `runMuonSel` only in the histograms it fills?

Reply:
> 

The THnSparse in runMuonSel has too many dimensions and the its large memory comsuption may lead to crash on hyperloop, so I add this lighter version to do certain analysis which does not need all the information.

Reply:
Fair enough, but that only highlights my previous comment.
You should use a configurable to decide which histograms are created in the `init` function and filled in the `runMuonSel` function instead of duplicating the code.

Reply:
> Fair enough, but that only highlights my previous comment. You should use a configurable to decide which histograms are created in the `init` function and filled in the `runMuonSel` function instead of duplicating the code.

Hi Vit, the changes has been made as you requested.